User Guide to Hamilton Standard (#187)
OpenMDAO/Aviary: aviary/docs/user_guide/hamilton_standard.md

# Hamilton Standard Propulsion Model

In the 1970s, NASA contracted Hamilton Standard to forecast into the future mid-80s to the 90s what they thought advanced propellers would look like. They took existing data and projected into the future. The result is what we call “Hamilton Standard model” today. The Hamilton Standard Documentation is publicly available on NTRS: [19720010354.pdf(https://ntrs.nasa.gov/api/citations/19720010354/downloads/19720010354.pdf) (nasa.gov). You can find the definitions, methodology, and Fortran code in the doc. In Aviary, we implement only one of the options, namely, for a given horsepower, it computes the corresponding thrust.

Below is an XDSM diagram of Hamilton Standard model:

![Hamilton Standard Diagram](images/hamilton_standard.png)

The inputs are grouped in three aspects:

Geometric inputs:
- Propeller diameter

> - Activity factor per blade (range: 80 to 200, baseline: 150)

Reviewer: Should "baseline" be "default"?

Author: Do you mean the word should be "default" or the default value should be 150? In Hamilton Standard, baseline means that the tables are provided based on that setting. I can change the actual default value to 150. Please let me know.

- Number of Blades (range: 2 to 8)

Power inputs:

> - Shaft Horsepower to Propeller (hp)

Reviewer: ```suggestion
- Shaft power to Propeller (hp)
```

Author: accepted.

- Installation Loss Factor (0 to 1)

Performance inputs:

> - Operating Altitude, ft
> - True Airspeed, knots

Reviewer: ```suggestion
- Operating Altitude (ft)
- True Airspeed (knots)
```

Author: Accept the suggestion.

> - Propeller Tip Speed (Usually < 800 ft/s) 

Reviewer: is this maximum or instantaneous?
If it is max as set by a user, make a note of it.
If it is just current from rpm*radius, no change is needed.

Author: It is not the maximum tip speed.

> - Integrated Lift Coefficient (range: 0.3 to 0.8, baseline: 0.5)

Reviewer: does "baseline" mean "default"

Reviewer: disregard, per response to other comment

Note that some of the inputs are good for limited ranges. The Hamilton Standard model can have odd number of blades although the data provided are based on even number of blades. For odd number of blades, interpolations using 2, 4, 6 and 8 blade data are used. The corresponding outputs are:

Geometric outputs:

> - Design Blade Pitch Angle (@0.75 Radius)

Reviewer: ```suggestion
- Design Blade Pitch Angle (at 0.75 Radius)
```

Author: agree. done.

- `USatmos`
- `PreHamiltonStandard`
- `HamiltonStandard`
- `InstallLoss`
- `PostHamiltonStandard`. 

`USatmos` component provides the flight condition. The flight condition is passed to `PreHamiltonStandard` component from which flight Mach number, propeller tip Mach number, advance ratio and power coefficient are computed. They are fed into `HamiltonStandard` component.

![CP and CT matching](images/CPE_CTE_matching.png)

HamiltonStandard is the core of the model. Given the power coefficient (CP) and advance ratio (J), it finds the blade angle (BL) by a CP-BL chart by tracing advance ratio and then with the blade angle, it finds the thrust coefficient (CT) using its CT-BL chart by tracing advance ratio again. This algorithm is shown in the above pair of charts. The CP → BL → CT chart matching algorithm is based on baseline data. If user inputs are not on baseline, it will first convert them to those baseline parameters by a sequence of interpolations to do corrections. The newly converted parameters are called “effective parameters” (e.g., CPE and CTE). The outputs are blade angle, thrust coefficient and tip compressibility loss factor.

> Finally, the thrust is computed in `PostHamiltonStandard` component based on thrust coefficient and tip compressibility loss factor.

Reviewer: ```suggestion
Finally, the thrust is computed in the `PostHamiltonStandard` component based on thrust coefficient and tip compressibility loss factor.
```

> Hamilton Standard model uses a bunch of wind tunnel test data from un-installed propellers. When a cell is mounted around the propeller, an installation loss factor is introduced. The installation loss factor can be given by user or computed. If it is computed, we need another group of components as shown below:

Reviewer: ```suggestion
The Hamilton Standard model uses wind tunnel test data from un-installed propellers. When a nacelle is mounted behind the propeller, an installation loss factor is introduced. The installation loss factor can be given by the user or computed. If it is computed, we need another group of components as shown below:
```

Author: Thanks for the modification. All are accepted.

![Installation Loss Factor](images/installation_loss_factor.png)

This diagram is represented by `InstallLoss` group in the first diagram. Note that nacelle diameter is needed when installation loss factor is computed. We use the average nacelle diameter.

The newly added aviary options and variables are:

```
Aircraft.Engine.PROPELLER_DIAMETER
Aircraft.Engine.PROPELLER_INTEGRATED_LIFT_COEFFICENT
Aircraft.Engine.PROPELLER_ACTIVITY_FACTOR
Aircraft.Engine.NUM_BLADES
Aircraft.Design.COMPUTE_INSTALLATION_LOSS
Dynamic.Mission.PROPELLER_TIP_SPEED
Dynamic.Mission.SHAFT_POWER
Dynamic.Mission.INSTALLATION_LOSS_FACTOR
```

> To build a turboprop engine and an Aviary model, we use `TurboPropDeck` object with a special parameter `prop_model` set to `True`:

Reviewer: ```suggestion
To build a turboprop engine that uses the Hamilton Standard propeller model, we use a `TurboPropDeck` object with `prop_model` set to `True`:
```

Author: updated.
